### PR TITLE
DO NOT MERGE YET: Refactor crisp

### LINKED ISSUE #5334: Errorbar: not aligned perfectly - sometimes has 1 pixel offset
#### Behaviour

Miss-aligned lines e.g. are not placed in center of linkedTo scatter markers.
#### Live demo with steps to reproduce

http://jsfiddle.net/zkenuwu2/
Plot lines added for reference, but also very much visible in compare to scatter markers.
#### Affected browser(s)

More visible in Chrome (at least for this demo).

### PR DESCRIPTION
**Pull request by TorsteinHonsi**
Fixed #5334, pixel alignment issues between grid lines, column series, error bars and scatter points.

This issue revealed the need for a full refactor of how crisping works, including for ticks/grid lines, scatter points and column series. The new aim is to implement a generic crisp function, so that every pixel position can be transformed to a reliable crisp position given the line width. This way, rendering elements from raw positions should always provide the same result.

### To do
 - [x] Fix clipping rectangle on all edges. Check #4504, #4656, #5146.
 - [x] Check inverted series. Probably crisping of whole numbers should be flipped.
 - [x] Pass unit tests
 - [x] Tick `reverseCrisp` should not be necessary
 - [x] Trawl the code for more places where the old crisping logic is in use
 - [x] Fix https://github.com/highcharts/highcharts/issues/20585 that became apparent when using non-rounded center positions for markers. https://jsfiddle.net/highcharts/3or2b4h9/.
 - [x] The rounding function argument for `crispLine` is only used twice. Check it. If it can be removed, `crispLine` and `crispPolyLine` can call `Utitlities.crisp` internally.
 - [x] Write test based on https://jsfiddle.net/highcharts/w7gpLsu1/
 - [x] To overcome the overwhelming visual diffs, postpone merge until immediately before the next release

### CODE REVIEW

File: samples/unit-tests/datalabels/multiple-labels/demo.js [hunk 1]
```diff
@@ -194,11 +194,13 @@ QUnit.test('Multiple data labels general tests.', function (assert) {
         reversed: false
     });
 
+    /* Fails after crisp refactor, but it's not a clear bug
```
Comment (KacperMadej) [inline samples/unit-tests/datalabels/multiple-labels/demo.js:197]: Looks like a bug here: bar type series' `dataLabels` with `inside: true` are showing for any y range - https://jsfiddle.net/BlackLabel/7stgvowh/1/

**Comment by KacperMadej** [comment]
Is the change for the hamburger menu icon intentional? (It shows well [here](http://utils.highcharts.local/samples/#test/highcharts/demo/renderer))

**Review by KacperMadej — COMMENTED** [review]
I looked through many different visual tests and all changes look like improvements except for a few issues I was able to find:

- Gantt's grid axis has a 1-pixel gap in the bottom right corner 
image

File: samples/unit-tests/datalabels/multiple-labels/demo.js (hunk 1 above)
Comment (TorsteinHonsi) [reply samples/unit-tests/datalabels/multiple-labels/demo.js:197]: Fixed

**Pull request merged**